komodo-core/src/main/java/org/komodo/repository/RepositoryImpl.java

```diff
@@ -433,31 +449,122 @@ public String decode(String encoded) {
     public static final String LIBRARY_ROOT = (KOMODO_ROOT + FORWARD_SLASH + Komodo.LIBRARY);
 
     /**
+     * The root path of the Komodo repository environment validation rules area
+     */
+    public static final String VALIDATION_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.VALIDATION;
+
+    /**
+     * The root path of the Komodo repository environment servers area
+     */
+    public static final String SERVERS_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.SERVERS;
+
+    /**
+     * The root path of the Komodo repository teiid cache area.
+     */
+    public static final String TEIID_CACHE_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.TEIID_CACHE;
+
+    protected static final KLog LOGGER = KLog.getLogger();
+
+    /**
      * The root path of the Komodo repository workspace area.
+     * This should remain private as clients should use
+     * {@link #komodoWorkspacePath(org.komodo.spi.repository.Repository.UnitOfWork)}
+     * in preference, allowing for the creation of the user home directory
      */
-    public static final String WORKSPACE_ROOT = (KOMODO_ROOT + FORWARD_SLASH + Komodo.WORKSPACE);
+    private static final String WORKSPACE_ROOT = (KOMODO_ROOT + FORWARD_SLASH + Komodo.WORKSPACE);
 
     /**
-     * The root path of the Komodo repository workspace searches area
+     * The root path of the Komodo repository workspace searches area.
+     * This remains a sibling of the user home directories so that searches can be shared by users.
      */
     public static final String SEARCHES_ROOT = WORKSPACE_ROOT + FORWARD_SLASH + Search.GROUP_NODE;
 
     /**
-     * The root path of the Komodo repository environment validation rules area
+     * @param transaction
+     *       the transaction (cannot be <code>null</code> or have a state that is not
+    *        {@link org.komodo.spi.repository.Repository.UnitOfWork.State#NOT_STARTED})
+    *
+     * @return true if the given transaction is a system transaction
      */
-    public static final String VALIDATION_ROOT = ENV_ROOT + FORWARD_SLASH + Environment.VALIDATION;
+    public static boolean isSystemTx(UnitOfWork transaction) {
+        if (transaction == null)
+            return false;
+
+        //
+        // Transactions should always have a user name but just in case one sneaked through
+        //
+        return transaction.getUserName() == null || SYSTEM_USER.equals(transaction.getUserName());
```

Comment:
I think you should get rid of the `null` check. Our transaction framework should guarantee the user name is never `null` so this check is not needed and might be hiding a real issue somewhere else.